src/include/function/udf_function.h

```diff
@@ -67,7 +67,7 @@ struct UDF {
     template<typename RESULT_TYPE>
     static function::scalar_func_exec_t createEmptyParameterExecFunc(RESULT_TYPE (*udfFunc)(),
         const std::vector<common::LogicalTypeID>&) {
-        (void*)(udfFunc); // Disable compiler warnings.
+        KU_UNUSED(udfFunc); // Disable compiler warnings.
```

Comment:
There's actually a built-in [[[maybe_unused]]](https://en.cppreference.com/w/cpp/language/attributes/maybe_unused) attribute that can be used instead. I'd added `KU_UNUSED` because I'd either forgotten about it or hadn't known about it. We should move to using that instead.

Reply:
Does `maybe_unused` also work with clang-tidy checks? If so, we should probably get rid of `KU_UNUSED` for good.

Reply:
I think so